pkg/plugins/approve/approve_test.go

```diff
@@ -1127,24 +1134,135 @@ Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a commen
+			files:               []string{"d/d.go"},
+			comments:            []*scm.Comment{},
+			reviews:             []*scm.Review{},
+			selfApprove:         false,
+			needsIssue:          false,
+			lgtmActsAsApprove:   false,
+			reviewActsAsApprove: false,
+			githubLinkURL:       &url.URL{Scheme: "https", Host: "github.com"},
+
+			expectDelete:  false,
+			expectToggle:  false,
+			expectComment: true,
+			expectedComment: `[APPROVALNOTIFIER] This PR is **NOT APPROVED**
+
+This pull-request has been approved by:
+The changes made require 2 more approval(s).
+To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **derek**
+You can assign the PR to them by writing ` + "`/assign @derek`" + ` in a comment when ready.
+
+The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=org%2Frepo).
+
+<details open>
+Needs approval from an approver in each of these files:
+
+- **[d/OWNERS](https://github.com/org/repo/blob/master/d/OWNERS)**
+
+Approvers can indicate their approval by writing ` + "`/approve`" + ` in a comment
+Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a comment
+</details>
+<!-- META={"approvers":["derek"]} -->`,
+		},
+		{
+			name:     "2 minimum reviewers - 1 approval",
+			hasLabel: false,
+			files:    []string{"d/d.go"},
+			comments: []*scm.Comment{
+				newTestComment("derek", "/approve"),
+			},
+			reviews:             []*scm.Review{},
+			selfApprove:         false,
+			needsIssue:          false,
+			lgtmActsAsApprove:   false,
+			reviewActsAsApprove: false,
+			githubLinkURL:       &url.URL{Scheme: "https", Host: "github.com"},
+
+			expectDelete:  false,
+			expectToggle:  false,
+			expectComment: true,
+			expectedComment: `[APPROVALNOTIFIER] This PR is **NOT APPROVED**
+
+This pull-request has been approved by: *[derek](<> "Approved")*
+The changes made require 1 more approval(s).
+To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
+You can assign the PR to them by writing ` + "`/assign `" + ` in a comment when ready.
+
+The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=org%2Frepo).
+
+<details open>
+Needs approval from an approver in each of these files:
+
+- ~~[d/OWNERS](https://github.com/org/repo/blob/master/d/OWNERS)~~ [derek]
+
+Approvers can indicate their approval by writing ` + "`/approve`" + ` in a comment
+Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a comment
+</details>
+<!-- META={"approvers":[]} -->`,
+		}, {
+			name:     "2 minimum reviewers - 2 approval",
+			hasLabel: false,
+			files:    []string{"d/d.go"},
+			comments: []*scm.Comment{
+				newTestComment("derek", "/approve"),
+				newTestComment("alice", "/approve"),
```

Comment:
This test should have failed since alice should not be counted as an approver for this file. As this is implemented now anybody that writes an '/approve' comment in the PR will be counted as an approver, without being being listed in OWNERS file. See
https://dashboard-jx.infra.jenkins-x.rocks/ns-jx/jenkins-x/lighthouse/PR-1638/1 for my test where changing `alice` to `anybody` still makes the test succeed.

Keep this test case, but expect the PR to not be approved. Add more test cases covering the case where there are the correct approvers. Preferably add a case with different files covered by different OWNERS as well.